### ISSUE TITLE
[Bug]: Color theme of fenced code blocks in MDX

### ISSUE THREAD

**Issue by StuDownie** [storybookjs/storybook#20639]
### Describe the bug

Can’t set the theme of fenced code blocks to dark in SB 7 beta

Previously, doing this in MDX would work:

\`\`\`javascript dark
	let foo;
\`\`\`

But all code blocks appear light now. 

Using a Source block works as expected:

`<Source language=“javascript” dark code={myCode} />`

### To Reproduce

```shell
Just add this to an MDX file to preview it:

 dark
	let foo;
```

### System

```shell
Environment Info:

  System:
    OS: macOS 13.1
    CPU: (4) x64 Intel(R) Core(TM) i5-8210Y CPU @ 1.60GHz
  Binaries:
    Node: 18.3.0 - /usr/local/opt/node@18/bin/node
    npm: 8.12.2 - /usr/local/opt/node@18/bin/npm
  Browsers:
    Chrome: 109.0.5414.87
    Safari: 16.2
  npmPackages:
    @storybook/addon-a11y: ^7.0.0-beta.28 => 7.0.0-beta.28 
    @storybook/addon-actions: ^7.0.0-beta.28 => 7.0.0-beta.28 
    @storybook/addon-essentials: ^7.0.0-beta.28 => 7.0.0-beta.28 
    @storybook/addon-interactions: ^7.0.0-beta.28 => 7.0.0-beta.28 
    @storybook/addon-links: ^7.0.0-beta.28 => 7.0.0-beta.28 
    @storybook/addon-postcss: ^2.0.0 => 2.0.0 
    @storybook/jest: ^0.0.11-next.0 => 0.0.11-next.0 
    @storybook/test-runner: ^0.10.0-next.4 => 0.10.0-next.4 
    @storybook/testing-library: ^0.0.14-next.1 => 0.0.14-next.1 
    @storybook/vue3: ^7.0.0-beta.28 => 7.0.0-beta.28 
    @storybook/vue3-vite: ^7.0.0-beta.28 => 7.0.0-beta.28
```

**Comment by nmpribeiro** [comment]
Same behaviour (all storybook addons 7.0.0-beta.33, nextjs 13, ts and sb7).

**Comment by JReinhold** [comment]
I'm surprised this has ever worked, any documentation I can find about MDX suggests that this needs the [`remark-mdx-code-meta`](https://github.com/remcohaszing/remark-mdx-code-meta) plugin in order for this to work. I can't find anywhere where it is mentioned that this works out of the box with MDX1 or 2, and I can't find anywhere in the code where we load this plugin.

https://mdxjs.com/guides/syntax-highlighting/#syntax-highlighting-with-the-meta-field

https://github.com/mdx-js/mdx/issues/702

@shilman do you have a clue here?

**Comment by shilman** [comment]
No idea. Happy to look at it together if it helps!

**Comment by imagoiq** [comment]
@sheriffMoose I understand from your PR that you want to stick to the default MDX2 behavior, which I strongly agree with. In the meantime, is there a way to get change the default theme for all code blocks from the preview config or anywhere else?

**Comment by xyy94813** [comment]
@imagoiq 
Hi, do you find any solution??

**Comment by imagoiq** [comment]
No, we stopped using fenced code blocks and used the `<Source>` element instead. And to keep code indentation and color for DX, we are storing code aside and importing it inside the `<Source>` element using the `?raw` vite feature like that: 
```
import SampleBreakpoints from './breakpoints.sample.scss?raw';

<Source
  code={SampleBreakpoints}
  language="scss"
/>
```

**Comment by xyy94813** [comment]
Try to control it with `parameters` failed

```ts
// failed
const meta = {
  parameters: {
    docs: {
      source: { dark: true },
    },
  },
};

export default meta;

// failed
export const Basic: Story = {
  parameters: {
    docs: {
      source: { dark: true },
    },
  },
};
```
```ts
// in .storybook/preview.ts 
export const parameters = {
  // failed
  doc: {
    source: {
      dark: false,
    }
  }
};
```

https://storybook.js.org/docs/react/api/doc-block-source#dark

**Comment by xyy94813** [comment]
https://github.com/storybookjs/storybook/blob/f18bd6fcbc1553b6dc60a9a5c6df88afcccbb1d0/code/addons/docs/src/DocsRenderer.tsx#L46-L49

Can i resolve it by override `components.code` ???

**Comment by StuDownie** [comment]
I've resorted to [downloading a Prism.js theme](https://github.com/PrismJS/prism-themes), forcing `!important` on every line, and including a bit at the top to override the div in Storybook doc:

```
[data-radix-scroll-area-viewport] {
    background-color: black !important;
}
```

Then importing that file in `preview.ts`

**Comment by nedkelly** [comment]
This doesn't look like it's ever been addressed, I'm using SB 7.5.3 and the docs addon 7.5.3, it's a little annoying because I'm trying to use fenced code blocks for "reasons" and in cases where the Source tag is not possible.

I have the preview parameters for dark theme enabled:
```javascript
docs: {
  source: {
    dark: true,
  },
},
```
And my results are mixed when using the fenced ` ```javascript dark` and `<Source dark ... />` 
![screenshot-localhost_6008-2023 12 01-11_20_56](https://github.com/storybookjs/storybook/assets/569534/ab0ac105-df1f-491b-b88f-cd3825402f61)

Basically the only place where the `dark` prop or option works is in imported Fenced code blocks in `.md` files, and if it's added as a prop on the `<Source />` component, all other ways fail. This is very inconsistent.

My needs are simply to make all code blocks dark no matter what. For now I'm also resorting to @StuDownie's [workaround](https://github.com/storybookjs/storybook/issues/20639#issuecomment-1831792046)!

**Comment by muditjuneja** [comment]
I was able to create a simple rehype plugin which allows you to pass properties like below -  
````
```mdx path=google.com src=no-src
# Heading 1
```
````

Later on this is available in the code component as a metaData prop which can be extracted like below - 

```
const components = {
  code: (props) => {
    console.log("props", props);
    const metaDataObject = props.metaData.split(' ').reduce((obj, item) => {
        const [key, value] = item.split('=');
        obj[key] = value;
        return obj;
      }, {});

    return <Code {...props} path={metaDataObject.path} src={metaDataObject.src} />;
  },
}
```
Using this in gatsby variant of Carbon design system here - https://github.com/carbon-design-system/gatsby-theme-carbon/pull/1289/files#diff-3c2d1eade54db26a66f65bf042875994c93c044734a9c62a19385e0d15435a35R132
